components/validator/guardian.go

```diff
@@ -189,21 +502,72 @@ func (g *Guardian) checkConfirmCondition(transactionId *big.Int, l2BlockNumber *
 	return true, nil
 }
 
+func (g *Guardian) updateInspectionWindow(start, end *big.Int) {
+	g.inspectionWindow = InspectionWindow{start: start, end: end}
+}
+
+// shouldBeDeleted checks the output should have been deleted or not.
+// It finds the output of the challenge that triggered the ReadyToProve event
+// and compares it to the local output of the guardian.
+func (g *Guardian) shouldBeDeleted(outputIndex *big.Int) (bool, error) {
+	readyToProveEvent := g.colosseumABI.Events[KeyEventReadyToProve]
+	addresses := []common.Address{g.cfg.ColosseumAddr}
+	eventIDTopic := []common.Hash{readyToProveEvent.ID}
+	outputIndexTopic := []common.Hash{common.BigToHash(outputIndex)}
+
+	query := ethereum.FilterQuery{
+		FromBlock: g.inspectionWindow.start,
+		ToBlock:   g.inspectionWindow.end,
+		Addresses: addresses,
+		Topics:    [][]common.Hash{eventIDTopic, outputIndexTopic},
+	}
+
+	logs, err := g.cfg.L1Client.FilterLogs(g.ctx, query)
+	if err != nil {
+		return false, fmt.Errorf("failed to get event logs related to outputs: %w", err)
+	}
+
+	if len(logs) == 0 {
+		return false, nil
+	}
+
+	cCtx, cCancel := context.WithTimeout(g.ctx, g.cfg.NetworkTimeout)
+	defer cCancel()
+	output, err := g.l2ooContract.GetL2Output(utils.NewSimpleCallOpts(cCtx), outputIndex)
+	if err != nil {
+		return false, fmt.Errorf("failed to get output from L2OutputOracle contract(outputIndex: %d): %w", outputIndex.Uint64(), err)
+	}
+
+	isValid, err := g.ValidateL2Output(g.ctx, output.OutputRoot, output.L2BlockNumber.Uint64())
+	if err != nil {
+		return false, err
+	}
+
+	return !isValid, nil
+}
+
 func (g *Guardian) ValidateL2Output(ctx context.Context, outputRoot eth.Bytes32, l2BlockNumber uint64) (bool, error) {
 	g.log.Info("validating deleted output as a result of challenge...", "l2BlockNumber", l2BlockNumber, "outputRoot", outputRoot)
 	localOutputRoot, err := g.OutputRootAtBlock(ctx, l2BlockNumber)
 	if err != nil {
 		return false, fmt.Errorf("failed to get output root at block number %d: %w", l2BlockNumber, err)
 	}
-	isEqual := bytes.Equal(outputRoot[:], localOutputRoot[:])
-	return isEqual, nil
+	isValid := bytes.Equal(outputRoot[:], localOutputRoot[:])
+	return isValid, nil
 }
 
 func (g *Guardian) ConfirmTransaction(ctx context.Context, transactionId *big.Int) (*types.Transaction, error) {
+	g.log.Info("crafting confirm tx", "transactionId", transactionId)
 	txOpts := utils.NewSimpleTxOpts(ctx, g.cfg.TxManager.From(), g.cfg.TxManager.Signer)
 	return g.securityCouncilContract.ConfirmTransaction(txOpts, transactionId)
 }
 
+func (g *Guardian) ForceDeleteOutput(ctx context.Context, outputIndex *big.Int) (*types.Transaction, error) {
```

Comment:
It does not delete output forcefully, just request deletion to security council members.
```suggestion
func (g *Guardian) RequestDeletion(ctx context.Context, outputIndex *big.Int) (*types.Transaction, error) {
```